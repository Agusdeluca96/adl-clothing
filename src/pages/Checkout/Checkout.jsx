import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import CheckoutItem from '../../components/CheckoutItem';
import { selectCartItems, selectCartTotal } from '../../redux/cart/cart.selector';

import './Checkout.scss';
import StripeCheckoutButton from '../../components/StripeCheckoutButton/StripeCheckoutButton';

const Checkout = ({ cartItems, cartTotal }) => (
  <div className="checkout">
    <div className="checkout-header">
      <div className="header-block">
        <span>Product</span>
      </div>
      <div className="header-block">
        <span>Description</span>
      </div>
      <div className="header-block">
        <span>Quantity</span>
      </div>
      <div className="header-block">
        <span>Price</span>
      </div>
      <div className="header-block">
        <span></span>
      </div>
    </div>
    {
      cartItems.map(cartItem =>
        <CheckoutItem key={cartItem.id} cartItem={cartItem} />
      )
    }
    <div className="checkout-footer">
      <div className="test-info">
        * Please use the following test credit card for payments *
      <br />
      Number: 4242 4242 4242 4242 - Exp: 04/20 - CVV: 123
    </div>
      <div className="total">
        <span>Total: ${cartTotal}</span>
        <StripeCheckoutButton price={cartTotal} />
      </div>
    </div>
  </div>
);

const mapStateToProps = createStructuredSelector({
  cartItems: selectCartItems,
  cartTotal: selectCartTotal
});

export default connect(mapStateToProps)(Checkout);