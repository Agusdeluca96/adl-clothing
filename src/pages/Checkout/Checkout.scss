.checkout {
  width: 80%;
  min-height: 90vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 50px auto 0;

  .checkout-header {
    width: 100%;
    padding: 10px 0;
    display: flex;
    justify-content: space-between;
    border-bottom: 1px solid #ecf0f1;

    .header-block {
      text-transform: capitalize;
      width: 20%;
      color: #ecf0f1;
      font-size: 20px;

      &:first-child {
        width: 15%;
      }
      &:nth-child(2) {
        width: 35%;
      }
      &:last-child {
        width: 10%;
      }
    }
  }

  .total {
    margin-top: 30px;
    margin-left: auto;
    font-size: 30px;
    color: #ecf0f1;
  }
}
