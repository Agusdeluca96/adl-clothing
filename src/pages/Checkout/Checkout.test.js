import React from 'react';
import { shallow } from 'enzyme';
import { Checkout } from './Checkout';

let wrapper;
beforeEach(() => {
  const mockProps = {
    cartItems: [],
    total: 100
  };

  wrapper = shallow(<Checkout {...mockProps} />);
});

it('should render Checkout page component', () => {
  expect(wrapper).toMatchSnapshot();
});
