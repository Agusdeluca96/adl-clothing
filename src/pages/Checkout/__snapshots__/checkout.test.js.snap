// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Checkout page component 1`] = `ShallowWrapper {}`;
